src/wrapper.jl

```diff
@@ -465,7 +467,38 @@ function LGBM_BoosterUpdateOneIter(bst::Booster)
     return is_finished[]
 end
 
-# function LGBM_BoosterUpdateOneIterCustom()
+"""
+LGBM_BoosterUpdateOneIterCustom
+Pass grads and 2nd derivatives corresponding to some custom loss function
+grads and 2nd derivatives must be same cardinality as training data
+Also, trying to run this on a booster without data will fail.
+"""
+function LGBM_BoosterUpdateOneIterCustom(bst::Booster, grads::Vector{<:AbstractFloat}, hessian::Vector{<:AbstractFloat})
+
+    if length(bst.datasets) == 0
+        throw(ErrorException("Booster does not have any training data associated"))
+    end
+    numdata = LGBM_DatasetGetNumData(first(bst.datasets))
+    nummodels = LGBM_BoosterNumModelPerIteration(bst)
+
+    if !((numdata*nummodels) == length(grads) == length(hessian))
+        throw(DimensionMismatch(
+            "Gradients sizes ($(length(grads)), $(length(hessian))) don't match training data size ($numdata) * ($nummodels)"
```

Comment:
number of trees per iteration * ($nummodels)

Reply:
Hmm? Not sure what you're expecting here, number of trees per iteration is the number of models, so the message would be like "doesnt match training data (22) * (3)" for example